daemon/libnetwork/config/config.go

```diff
@@ -162,3 +165,22 @@ func OptionFirewallBackend(val string) Option {
 		c.FirewallBackend = val
 	}
 }
+
+// OptionRootless returns an option setter that indicates whether the daemon is
+// running in rootless mode.
+func OptionRootless(rootless bool) Option {
+	return func(c *Config) {
+		if rootless {
+			c.Rootless = rootless
+		}
+	}
+}
```

Comment:
Why is `OptionRootless(false)` a no-op? All the other options are last-set-wins; why the special case where `OptionRootless(false), OptionRootless(true)` and `OptionRootless(true), OptionRootless(false)` both set rootless mode? While it's not going to cause much trouble in the daemon proper given that the rootless-mode state of a dockerd process is immutable, it is a violation of the principle of least astonishment and a gotcha which has the potential to frustrate test authors.

```suggestion
func OptionRootless(rootless bool) Option {
	return func(c *Config) {
		c.Rootless = rootless
	}
}
```